apps/frontend/src/app/PageTeam/CharacterDisplay/Build/EquipBuildModal.tsx

```diff
@@ -0,0 +1,213 @@
+import type { ArtifactSlotKey } from '@genshin-optimizer/gi/consts'
+import { useDatabase } from '@genshin-optimizer/gi/db-ui'
+import { getCharData } from '@genshin-optimizer/gi/stats'
+import { Checkroom, KeyboardArrowDown } from '@mui/icons-material'
+import {
+  Box,
+  Button,
+  CardContent,
+  CardHeader,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  Grid,
+  TextField,
+  Typography,
+} from '@mui/material'
+import { useContext, useState } from 'react'
+import ArtifactCardNano from '../../../Components/Artifact/ArtifactCardNano'
+import CloseButton from '../../../Components/CloseButton'
+import WeaponCardNano from '../../../Components/Weapon/WeaponCardNano'
+import { TeamCharacterContext } from '../../../Context/TeamCharacterContext'
+
+type EquipChangeProps = {
+  currentName: string
+  currentWeapon: string | undefined
+  currentArtifacts: Record<ArtifactSlotKey, string | undefined>
+  newWeapon: string | undefined
+  newArtifacts: Record<ArtifactSlotKey, string | undefined>
+}
+
+export default function EquipBuildModal({
+  equipChangeProps,
+  showPrompt,
+  OnHidePrompt,
+  onEquip,
+}: {
+  equipChangeProps: EquipChangeProps
+  showPrompt: boolean
+  onEquip: () => void
+  OnHidePrompt: () => void
+}) {
+  const [name, setName] = useState('')
+  const [copyCurrent, setCopyCurrent] = useState(false)
+  // const [showPrompt, onShowPrompt, OnHidePrompt] = useBoolState()
+
+  const database = useDatabase()
+  const {
+    teamCharId,
+    teamChar: { key: characterKey },
+  } = useContext(TeamCharacterContext)
+  const weaponTypeKey = getCharData(characterKey).weaponType
+
+  const toEquip = () => {
+    if (copyCurrent) {
+      database.teamChars.newBuild(teamCharId, {
+        name:
+          name !== '' ? name : `Duplicate of ${equipChangeProps.currentName}`,
+        artifactIds: equipChangeProps.currentArtifacts,
+        weaponId: equipChangeProps.currentWeapon,
+      })
+    }
+
+    onEquip()
+    setName('')
+    setCopyCurrent(false)
+    OnHidePrompt()
+  }
+
+  /* TODO: Dialog Wanted to use a Dialog here, but was having some weird issues with closing out of it */
+  /* TODO: Translation */
+  return (
+    <ModalWrapper open={showPrompt} onClose={OnHidePrompt}>
+      <CardThemed>
+        <CardHeader
+          title={
+            <Box sx={{ display: 'flex', gap: 1, alignItems: 'center' }}>
+              <Checkroom />
+              <span>Confirm Equipment Changes</span>
+            </Box>
+          }
+          action={<CloseButton onClick={OnHidePrompt} />}
+        />
+        <Divider />
+        <CardContent
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+          }}
+        >
+          <Box>
+            {/* Active Build */}
+            <CardThemed bgt="light">
+              <CardHeader
+                sx={{ pb: 0 }}
+                title={`${equipChangeProps.currentName}: Current Equipment`}
```

Comment:
I'd suggest using a badge to show information:
`<Sqbadgecolor="warning">Current Equipment</Sqbadge>`

Reply:
done